src/Microsoft.AspNet.Authentication/AuthenticationMiddleware.cs

```diff
@@ -41,32 +37,29 @@ public abstract class AuthenticationMiddleware<TOptions> where TOptions : Authen
 
         public async Task Invoke(HttpContext context)
         {
-            using (RequestServicesContainer.EnsureRequestServices(context, _services))
+            AuthenticationHandler<TOptions> handler = CreateHandler();
+            await handler.Initialize(Options, context);
+            try
+            {
+                if (!await handler.InvokeAsync())
+                {
+                    await _next(context);
+                }
+            }
+            catch (Exception)
             {
-                AuthenticationHandler<TOptions> handler = CreateHandler();
-                await handler.Initialize(Options, context);
                 try
                 {
-                    if (!await handler.InvokeAsync())
-                    {
-                        await _next(context);
-                    }
+                    handler.Faulted = true;
```

Comment:
On a side note, that could be useful to store the `Exception` caught instead of just having a `Faulted` property.